PostBoard.Api/Services/BirthdayService.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using PostBoard.Api.Models;
+using PostBoard.Api.Services.Contracts;
+
+namespace PostBoard.Api.Services;
+
+public class BirthdayService : IBirthdayService
+{
+    private readonly PostBoardContext _dbContext;
+
+    public BirthdayService(PostBoardContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+ 
+    public async Task<IList<Birthday>> GetAllAsync()
+    {
+        return await _dbContext.Birthdays.ToListAsync();
+    }
+
+    public async Task<Birthday> GetByIdAsync(int id)
+    {
+        return await _dbContext.Birthdays.FirstOrDefaultAsync(p => p.Id == id);
+    }
+
+    public async Task<int> CreateAsync(Birthday input)
+    {
+        _dbContext.Birthdays.Add(input);
+        await _dbContext.SaveChangesAsync();
+        
+        return input.Id;
+    }
+
+   
+
+    public async Task UpdateAsync(int id,
+        Birthday input)
+    {
+        var birthday = await _dbContext.Birthdays.FirstOrDefaultAsync(p => p.Id == id);
+        if (birthday == null)
+        {
+            throw new InvalidOperationException("Birthday not found");
```

Comment:
Replace `InvalidOperationException` with just `Exception`. Later we will modify it somehow. But good job! `InvalidOperationException` is good idea.